### PR TITLE
[opentelemetry-operator] update CRDs

### PR DESCRIPTION
**Pull request by timojohlo**
## Pull Request Details

-  add appVersion
- update CRDs with correct source: https://github.com/open-telemetry/opentelemetry-operator/tree/main/config/crd/bases
(instead of: https://github.com/open-telemetry/opentelemetry-helm-charts/tree/main/charts/opentelemetry-operator/conf/crds)

### CODE REVIEW

File: opentelemetry-operator/chart/crds/crd-opentelemetrycollector.yaml [hunk 1]
```diff
@@ -21,7 +21,6 @@ spec:
           namespace: {{ .Release.Namespace }}
           path: /convert
           port: {{ .Values.admissionWebhooks.servicePort }}
-  {{ if .caBundle }}{{ cat "caBundle:" .caBundle | indent 8 }}{{ end }}
```
Comment (viennaa) [inline opentelemetry-operator/chart/crds/crd-opentelemetrycollector.yaml:24]: Why is this not needed anymore?

Comment (timojohlo) [reply opentelemetry-operator/chart/crds/crd-opentelemetrycollector.yaml:24]: We are not using the caBundle right now and this is currently hindering helm to correctly render the chart.

Comment (viennaa) [reply opentelemetry-operator/chart/crds/crd-opentelemetrycollector.yaml:24]: Is this planned to be used in the future? 

If so, add an empty directive in your values and the `if` will evaluate to false.

```yaml
opentelemetry-operator
   [...]
   caBundle:
   [...]
```

Comment (timojohlo) [reply opentelemetry-operator/chart/crds/crd-opentelemetrycollector.yaml:24]: Not sure, but I agree with you that I make sense to make this optional. The reason I removed it was that the line
`  {{ if .caBundle }}{{ cat "caBundle:" .caBundle | indent 8 }}{{ end }}
`
is afaik not valid in helm and the function `cat` throws an error when installing the chart.

Comment (timojohlo) [reply opentelemetry-operator/chart/crds/crd-opentelemetrycollector.yaml:24]: the source for the CRDs were wrong. The correct source for CRDs is https://github.com/open-telemetry/opentelemetry-operator/tree/main/config/crd/bases

File: opentelemetry-operator/chart/Chart.yaml [hunk 2]
```diff
@@ -2,8 +2,9 @@
 # SPDX-License-Identifier: Apache-2.0
 
 apiVersion: v2
+appVersion: 0.104.0
```
Comment (viennaa) [inline opentelemetry-operator/chart/Chart.yaml:5]:
```suggestion
appVersion: v0.104.0
```

I'd name it exactly like the version upstream.

**Review by viennaa — CHANGES_REQUESTED** [review]

**Review by viennaa — APPROVED** [review]
Thanks, let's go :)

**Pull request merged**